py/picca/fitter2/pk.py

```diff
@@ -36,45 +44,35 @@ def pk_gauss_smoothing(k, pk_lin, tracer1, tracer2, **kwargs):
     sp2 = kwargs['par_sigma_smooth']**2
     return sp.exp(-(kp**2*sp2+kt**2*st2)/2.)
 
-def pk_kaiser(k, pk_lin, tracer1, tracer2, **kwargs):
-    bias1, beta1, bias2, beta2 = bias_beta(kwargs, tracer1, tracer2)
-    pk = bias1*bias2*pk_lin*(1+beta1*muk**2)*(1+beta2*muk**2)
-
-    return pk
-
-def pk_hcd(k, pk_lin, tracer1, tracer2, **kwargs):
-
-    bias1, beta1, bias2, beta2 = bias_beta(kwargs, tracer1, tracer2)
-
-    bias_hcd = kwargs["bias_hcd"]
-    beta_hcd = kwargs["beta_hcd"]
-    L0 = kwargs["L0_hcd"]
+def pk_gauss_exp_smoothing(k, pk_lin, tracer1, tracer2, **kwargs):
+    """
+    Apply a Gaussian and exp smoothing to the full correlation function (use full for london_mocks_v6.0
 
-    kp = k*muk
-    F_hcd = utils.sinc(kp*L0)
+    """
+    kp  = k*muk
+    kt  = k*sp.sqrt(1.-muk**2)
+    st2 = kwargs['per_sigma_smooth']**2
+    sp2 = kwargs['par_sigma_smooth']**2
 
-    bias_eff1 = bias1 + bias_hcd*F_hcd
-    beta_eff1 = (bias1 * beta1 + bias_hcd*beta_hcd*F_hcd)/(bias1 + bias_hcd*F_hcd)
+    et2 = kwargs['per_exp_smooth']**2
+    ep2 = kwargs['par_exp_smooth']**2
 
-    bias_eff2 = bias2 + bias_hcd*F_hcd
-    beta_eff2 = (bias2 * beta2 + bias_hcd*beta_hcd*F_hcd)/(bias2 + bias_hcd*F_hcd)
+    return sp.exp(-(kp**2*sp2+kt**2*st2)/2.)*sp.exp(-(np.abs(kp)*ep2+np.abs(kt)*et2) )
 
-    pk = pk_lin*bias_eff1*bias_eff2*(1 + beta_eff1*muk**2)*(1 + beta_eff2*muk**2)
+def pk_kaiser(k, pk_lin, tracer1, tracer2, **kwargs):
+    bias1, beta1, bias2, beta2 = bias_beta(kwargs, tracer1, tracer2)
+    pk = bias1*bias2*pk_lin*(1+beta1*muk**2)*(1+beta2*muk**2)
 
     return pk
 
-def pk_hcd_Rogers2018(k, pk_lin, tracer1, tracer2, **kwargs):
-    """Model the effect of HCD systems with the Fourier transform
-       of a Lorentzian profile. Motivated by Rogers et al. (2018).
-
-    Args:
-        Same than pk_hcd
+def pk_hcd(k, pk_lin, tracer1, tracer2, **kwargs):
+    """
+    Use Fvoigt function to fit the DLA in the autocorrelation Lyman-alpha without masking them ! (L0 = 1)
 
-    Returns:
-        Same than pk_hcd
+    (If you want to mask them --> use Fvoigt_exp.txt and L0 = 10 as eBOOS DR14)
 
     """
-
+    global Fvoigt_data
```

Comment:
could you explain why you need to do that?

Reply:
I need a global variable here because I need to use it the function pk_hcd and I don't manage to pass it in the argument of this function... So I declare Fvoigt_data in the initialization of the class and I can use this variable in the function pk_hcd when I call pk_hcd at any moment